dGame/dComponents/BaseCombatAIComponent.h

```diff
@@ -19,7 +19,7 @@ class Entity;
 /**
  * The current state of the AI
  */
-enum class AiState : int {
+enum class AiState : uint32_t {
```

Comment:
Keep this as an int

Reply:
Why? Isn't the state serialized as a int32_t anyway?

Reply:
Yeah I get what you mean now, feel free to add back in this change.